astropy/modeling/fitting.py

```diff
@@ -1240,15 +1240,26 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
                     for _ in weights * np.array(model.fit_deriv(x, y, *params))
                 ]
 
-    def _compute_param_cov(self, model, y, init_values, cov_x, fitparams, farg):
+    def _compute_param_cov(
+        self, model, y, init_values, cov_x, fitparams, farg, weights=None
+    ):
         # now try to compute the true covariance matrix
         if (len(y) > len(init_values)) and cov_x is not None:
-            sum_sqrs = np.sum(self.objective_function(fitparams, *farg) ** 2)
-            dof = len(y) - len(init_values)
-            self.fit_info["param_cov"] = cov_x * sum_sqrs / dof
+            self.fit_info["param_cov"] = cov_x
+            if weights is None:
```

Comment:
Should we also give the option to toggle the behavior like the `absolute_sigma` option that `scipy.optimize.curve_fit` gives users for the case when no weights are passed?

Reply:
I understand giving people options is generally a good thing, and the generic name of the kwarg is `weights`. But we say in the docstring that `weights = 1 / sigma`, which is _not_ true if `absolute_sigma == False`.

Reply:
Right, so then the question becomes "should we add an option to disable this change". That is add an option which allows users to return the behavior of the weights to how they previously worked? My concern here lies with the fact that this change will technically change this computation's output in some cases. This can have knock-on effects for downstream users, who may wish to return to the previous behavior.

@pllim what is your opinion here?

Reply:
> That is add an option which allows users to return the behavior of the weights to how they previously worked?

Unless the old way is completely wrong, it is generally good to provide backward compatibility.

Reply:
I would argue the old way is completely wrong. We offered a keyword to adjust the weights, but did not compute the result based on the weights.

Reply:
Also – the user can have "perfect" backwards compatibility _if_ they set `weights = None`.

Reply:
As I told Brett offline, API change is acceptable as long as it is not backported. And if an extra option lets people get the old results, then fine by me but ultimately it is up to @astropy/modeling .

Reply:
In principle, I am okay with this change so long as it is not backported. However, we should probably advertise the change with a "whats-new". So that it is clear when reading the changes for astropy 5.3 that this behavior will be slightly different.